lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -342,31 +350,20 @@ private void writeFooter()
         createDataOutput(MAGIC).writeData(outputStream);
     }
 
-    Slice getFooter(List<RowGroup> rowGroups, MessageType messageType)
-            throws IOException
-    {
-        FileMetaData fileMetaData = new FileMetaData();
-        fileMetaData.setVersion(1);
-        fileMetaData.setCreated_by(createdBy);
-        fileMetaData.setSchema(MessageTypeConverter.toParquetSchema(messageType));
-        // Added based on org.apache.hadoop.hive.ql.io.parquet.write.DataWritableWriteSupport
-        parquetTimeZone.ifPresent(dateTimeZone -> fileMetaData.setKey_value_metadata(
-                ImmutableList.of(new KeyValue("writer.time.zone").setValue(dateTimeZone.getID()))));
-        long totalRows = rowGroups.stream().mapToLong(RowGroup::getNum_rows).sum();
-        fileMetaData.setNum_rows(totalRows);
```

Comment:
where `setNum_rows` went to?

the new code doesn't apparently call this method?

Reply:
It became a constructor argument to FileMetadata